frontend/src/components/ChakraDataTable.tsx

```diff
@@ -494,52 +557,78 @@ export const ChakraDataTable = <T extends { id: number }>({
                               header.getContext(),
                             )}
                       </div>
-                      {features?.sorting && (
+                      {features?.sorting && header.column.getCanSort() && (
                         <SortIcon isSorted={header.column.getIsSorted()} />
                       )}
-                    </HStack>
-                  </Table.ColumnHeader>
+                    </HeaderContent>
+                  </ColumnHeader>
                 ))}
-              </Table.Row>
+              </TableRow>
             ))}
           </Table.Header>
 
           <Table.Body>
             {table.getRowModel().rows.map((row) => (
-              <Table.Row
+              <TableRow
                 key={row.id}
                 onClick={() => onRowClick?.(row.original)}
-                style={{ cursor: onRowClick ? "pointer" : "default" }}
+                className={rowClassName}
               >
                 {row.getVisibleCells().map((cell) => (
-                  <Table.Cell key={cell.id}>{cellWrapper({ cell })}</Table.Cell>
+                  <StyledTableCell key={cell.id}>
+                    {cellWrapper({ cell })}
+                  </StyledTableCell>
                 ))}
-              </Table.Row>
+              </TableRow>
             ))}
           </Table.Body>
         </Table.Root>
 
-        {features?.pagination && (
-          <HStack justify="center" gap={2}>
-            <Button
-              onClick={() => table.previousPage()}
+        {features?.pagination && table.getPageCount() > 1 && (
+          <PaginationWrapper>
+            <PaginationIconButton
+              aria-label="Previous page"
+              onClick={(e) => {
+                e.stopPropagation();
+                table.previousPage();
+              }}
               disabled={!table.getCanPreviousPage()}
+              variant="ghost"
```

Comment:
Can we highlight the current page? Also why not use https://chakra-ui.com/docs/components/pagination directly that offers all we want I think?